src/CMakeLists.txt

```diff
@@ -69,17 +69,38 @@ if ("${JUPYTER}" STREQUAL "ON")
    add_definitions(-DRB_XCODE)
 endif()
 
+#MESSAGE("Boost location hints:")
+#if (BOOST_ROOT)
+#  MESSAGE("  BOOST_ROOT: ${BOOST_ROOT}")
+#else()
+#  MESSAGE("  BOOST_ROOT not set")
+#endif()
+#if (BOOST_INCLUDEDIR)
+#  MESSAGE("  BOOST_INCLUDEDIR: ${BOOST_INCLUDEDIR}")
+#endif()
+#if (BOOST_LIBRARYDIR)
+#  MESSAGE("  BOOST_LIBRARYDIR: ${BOOST_LIBRARYDIR}")
+#endif()
 MESSAGE("Boost location hints:")
+if (BOOST_ROOT OR BOOST_INCLUDEDIR OR BOOST_LIBRARYDIR) 
+  set(Boost_NO_SYSTEM_PATHS ON)
+  set(Boost_NO_BOOST_CMAKE ON)
+  set(Boost_NO_WARN_NEW_VERSION ON)
+endif()
 if (BOOST_ROOT)
   MESSAGE("  BOOST_ROOT: ${BOOST_ROOT}")
-else()
-  MESSAGE("  BOOST_ROOT not set")
+  set(Boost_INCLUDE_DIR ${BOOST_ROOT}/include)
+  set(Boost_LIBRARY_DIR ${BOOST_ROOT}/lib)
+  INCLUDE_DIRECTORIES(${Boost_INCLUDE_DIR})
 endif()
 if (BOOST_INCLUDEDIR)
   MESSAGE("  BOOST_INCLUDEDIR: ${BOOST_INCLUDEDIR}")
+  set(Boost_INCLUDE_DIR ${BOOST_ROOT})
+  INCLUDE_DIRECTORIES(${Boost_INCLUDE_DIR})
```

Comment:
Why do we need to add an `INCLUDE_DIRECTORIES(...)` flag directly?
Doesn't cmake do this already?

Reply:
I do not quite remember the rationale for this. I will work with it tomorrow to figure it out.

Reply:
Just tried removing this and tested on a couple of machines. On my laptop (nixos) it does not seem to be needed. However, on a debian cloud instance removing it results in the boost components not being found. I think maybe disabling the CMake specific boost settings changes how the includes are handled internally. 

I don't have access to a machine that has boost installed by the system in a normal location (i.e. an install that allows for just using `./build.sh` without any flags), so I can not test this theory though.